tests/core/test_collections.py

```diff
@@ -101,3 +103,28 @@ def test_collection_defaults_to_hook_config():
     hook._prepare_collections()
     assert cm.get("foo").save_config.mode_save_configs[ModeKeys.TRAIN].save_interval == 10
     assert cm.get("foo").reduction_config.save_raw_tensor is True
+
+
+def test_invalid_collection_config_exception():
```

Comment:
are there any failures in the existing tests? like the rules tests and scalar tests?

Reply:
There were failures in the rules repository, which were addressed in a  separate [PR](https://github.com/awslabs/sagemaker-debugger-rules/pull/194)